### PR TITLE
feat(route): add Apache APISIX 博客 & 云原生社区博客 & 支流科技博客

### PR DESCRIPTION
**Pull request by aneasystone**
## 该 PR 相关 Issue / Involved Issue

Close #

## 路由地址示例 / Example for the Proposed Route(s)

```routes
/apache/apisix/blog
/cloudnative/blog
/apiseven/blog
```

## 新 RSS 路由检查表 / New RSS Route Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [ ] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [x] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/v2/apache/apisix/blog.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        const time = $(elem).find('footer').find('time').text();
+        const author = $(elem).find('footer').find('img').attr('src');
+        results.push({
+            title: a.find('h2').text(),
+            description: a.find('p').text(),
+            link: a.attr('href'),
+            pubDate: timezone(parseDate(time, 'YYYY年M月D日'), +8),
+            author,
+        });
+    });
```
Comment (TonyRL) [inline lib/v2/apache/apisix/blog.js:23]: Please use [map](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array/map) to create a new array.

File: lib/v2/apache/apisix/blog.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    return results;
+}
+
+module.exports = async (ctx) => {
+    const articles = await getArticles();
+    ctx.state.data = {
+        title: `Blog | Apache APISIX`,
+        link: `https://apisix.apache.org/zh/blog/`,
```
Comment (TonyRL) [inline lib/v2/apache/apisix/blog.js:31]:
```suggestion
        title: 'Blog | Apache APISIX',
        link: 'https://apisix.apache.org/zh/blog/',
```

File: lib/v2/apiseven/blog.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+const { parseDate } = require('@/utils/parse-date');
+const timezone = require('@/utils/timezone');
+
+function getDetail(link, ctx) {
+    return ctx.cache.tryGet(
+        link,
+        async () => {
+            const { data: res } = await got(link);
+            const $ = cheerio.load(res);
+            const json = JSON.parse($('#__NEXT_DATA__').html());
```
Comment (TonyRL) [inline lib/v2/apiseven/blog.js:12]:
```suggestion
            const json = JSON.parse($('#__NEXT_DATA__').text());
```

File: lib/v2/apiseven/blog.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        },
+        60 * 60 * 24 * 7
+    );
+}
+
+async function getArticles() {
+    const url = 'https://www.apiseven.com/blog';
+    const { data: res } = await got(url);
+    const $ = cheerio.load(res);
+    const json = JSON.parse($('#__NEXT_DATA__').html());
```
Comment (TonyRL) [inline lib/v2/apiseven/blog.js:25]:
```suggestion
    const json = JSON.parse($('#__NEXT_DATA__').text());
```

File: lib/v2/apiseven/blog.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+                description: item.description,
+                link: item.link,
+                pubDate: item.pubDate,
+                author: detail.author,
+            };
+        })
+    );
+
+    ctx.state.data = {
+        title: `博客 | 支流科技`,
```
Comment (TonyRL) [inline lib/v2/apiseven/blog.js:50]:
```suggestion
        title: '博客 | 支流科技',
```

File: lib/v2/apiseven/blog.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        link: `https://www.apiseven.com/blog`,
```
Comment (TonyRL) [inline lib/v2/apiseven/blog.js:51]:
```suggestion
        link: 'https://www.apiseven.com/blog',
```

File: lib/v2/apiseven/blog.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        link,
+        async () => {
+            const { data: res } = await got(link);
+            const $ = cheerio.load(res);
+            const json = JSON.parse($('#__NEXT_DATA__').html());
+            return {
+                author: json.props.pageProps.post.author_name,
+            };
+        },
+        60 * 60 * 24 * 7
```
Comment (TonyRL) [inline lib/v2/apiseven/blog.js:17]: The default value of `CACHE_CONTENT_EXPIRE` is already long enough.

File: lib/v2/cloudnative/blog.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+        const meta = $(elem).find('.stream-meta .article-metadata');
+        const time = meta.find('.article-date').text().replace('发布于', '');
+        results.push({
+            title: a.text(),
+            link: a.attr('href'),
+            description: summary.text(),
+            pubDate: timezone(parseDate(time, 'YYYY-MM-DD'), +8),
+            author: meta.find('span').eq(0).find('a').text(),
+        });
+    });
```
Comment (TonyRL) [inline lib/v2/cloudnative/blog.js:24]: Please use map to create a new array.

File: lib/v2/cloudnative/blog.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    return results;
+}
+
+module.exports = async (ctx) => {
+    const articles = await getArticles();
+    ctx.state.data = {
+        title: `博客 | 云原生社区（中国）`,
+        link: `https://cloudnative.to/blog/`,
```
Comment (TonyRL) [inline lib/v2/cloudnative/blog.js:32]:
```suggestion
        title: '博客 | 云原生社区（中国）',
        link: 'https://cloudnative.to/blog',
```

File: lib/v2/apache/apisix/blog.js [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    const articles = $('section.sec_gjjg').eq(1).find('article');
+    return articles.toArray().map((elem) => {
+        const a = $(elem).find('header > a');
+        const time = $(elem).find('footer').find('time').text();
+        const author = $(elem).find('footer').find('img').attr('src');
+        return {
+            title: a.find('h2').text(),
+            description: a.find('p').text(),
+            link: a.attr('href'),
+            pubDate: timezone(parseDate(time, 'YYYY年M月D日'), +8),
```
Comment (TonyRL) [inline lib/v2/apache/apisix/blog.js:19]: Please use `datetime` attributes instead.

File: lib/v2/apache/apisix/blog.js [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+            author,
+        };
```
Comment (TonyRL) [inline lib/v2/apache/apisix/blog.js:21]:
```suggestion
            category: $(elem).find('header div a').toArray().map((elem) => $(elem).text()),
        };
```

File: lib/v2/apiseven/blog.js [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+async function getArticles() {
+    const url = 'https://www.apiseven.com/blog';
+    const { data: res } = await got(url);
+    const $ = cheerio.load(res);
+    const json = JSON.parse($('#__NEXT_DATA__').text());
+    return json.props.pageProps.list.map((item) => ({
+        title: item.title,
+        description: item.title,
+        link: 'https://www.apiseven.com' + item.slug,
+        pubDate: timezone(parseDate(item.published_at), +8),
```
Comment (TonyRL) [inline lib/v2/apiseven/blog.js:26]:
```suggestion
        pubDate: timezone(parseDate(item.published_at), +8),
        category: item.tags,
```

File: lib/v2/apiseven/blog.js [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+const { parseDate } = require('@/utils/parse-date');
+const timezone = require('@/utils/timezone');
+
+function getDetail(link, ctx) {
+    return ctx.cache.tryGet(link, async () => {
+        const { data: res } = await got(link);
+        const $ = cheerio.load(res);
+        const json = JSON.parse($('#__NEXT_DATA__').text());
+        return {
+            author: json.props.pageProps.post.author_name,
```
Comment (TonyRL) [inline lib/v2/apiseven/blog.js:12]: description is also presented in `json.props.pageProps.post.content`, please add it as well,

File: lib/v2/cloudnative/blog.js [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+        const a = $(elem).find('.article-title > a');
+        const summary = $(elem).find('.summary-link');
+        const meta = $(elem).find('.stream-meta .article-metadata');
+        const time = meta.find('.article-date').text().replace('发布于', '');
+        return {
+            title: a.text(),
+            link: a.attr('href'),
+            description: summary.text(),
+            pubDate: timezone(parseDate(time, 'YYYY-MM-DD'), +8),
+            author: meta.find('span').eq(0).find('a').text(),
```
Comment (TonyRL) [inline lib/v2/cloudnative/blog.js:21]:
```suggestion
            author: meta.find('span').eq(0).find('a').text(),
            category: meta.find('.article-categories a').text(),
```

File: lib/v2/apache/radar.js [hunk 15]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+    'apache.org': {
+        _name: 'Apache',
+        apisix: [
+            {
+                title: 'APISIX 博客',
+                docs: 'https://docs.rsshub.app/blog.html#apache',
+                source: ['/zh/blog'],
+                target: () => `/apache/apisix/blog`,
```
Comment (TonyRL) [inline lib/v2/apache/radar.js:9]:
```suggestion
                target: () => '/apache/apisix/blog',
```

File: lib/v2/apiseven/radar.js [hunk 16]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+    'apiseven.com': {
+        _name: '支流科技',
+        '.': [
+            {
+                title: '博客',
+                docs: 'https://docs.rsshub.app/blog.html#zhi-liu-ke-ji',
+                source: ['/blog'],
+                target: () => `/apiseven/blog`,
```
Comment (TonyRL) [inline lib/v2/apiseven/radar.js:9]:
```suggestion
                target: () => '/apiseven/blog',
```

File: lib/v2/cloudnative/radar.js [hunk 17]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+    'cloudnative.to': {
+        _name: '云原生社区',
+        '.': [
+            {
+                title: '博客',
+                docs: 'https://docs.rsshub.app/blog.html#yun-yuan-sheng-she-qu',
+                source: ['/blog'],
+                target: () => `/cloudnative/blog`,
```
Comment (TonyRL) [inline lib/v2/cloudnative/radar.js:9]:
```suggestion
                target: () => '/cloudnative/blog',
```

File: lib/v2/apache/apisix/blog.js [hunk 18]
```diff
@@ -0,0 +1,32 @@
...
+    return articles.toArray().map((elem) => {
+        const a = $(elem).find('header > a');
+        const time = $(elem).find('footer').find('time').text();
+        const author = $(elem).find('footer').find('img').attr('src');
+        return {
+            title: a.find('h2').text(),
+            description: a.find('p').text(),
+            link: a.attr('href'),
+            pubDate: timezone(parseDate(time, 'YYYY年M月D日'), +8),
+            author,
```
[the remark this answers is not in the document]
Comment (aneasystone) [reply lib/v2/apache/apisix/blog.js]: the `author` attribute is removed

File: lib/v2/apiseven/blog.js [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+    const items = await Promise.all(
+        articles.map(async (item) => {
+            const detail = await getDetail(item.link, ctx);
+            return {
+                title: item.title,
+                description: detail.description,
+                link: item.link,
+                pubDate: item.pubDate,
+                author: detail.author,
+            };
```
Comment (TonyRL) [inline lib/v2/apiseven/blog.js:42]: Please cache these operations instead of `getDetail`.

**Pull request merged**